### DISCUSSION TITLE
Transition issued with comment

### DISCUSSION THREAD

**Discussion by azamora-zsc** [ankitpokhrel/jira-cli#472] [Ideas] [answered]
Right now I can't use the command move to transition issues in my project. It's enforced for us to add a comment when we move an issue to certain statuses (for instance, Done) so I get this:
```
$ jira issue move ISS-5516 "Done"

Error:
  - comment: Comment is required.

jira: Received unexpected response '400 '.
Please check the parameters you supplied and try again.
```
Adding a flag to the move command to add a comment as part of the transition action would help us. Something like:
```
$ jira issue move ISS-3245 "Done" -c "My comment about the transition"
```
Thanks for such a great tool, by the way.

**Comment by ankitpokhrel** [comment]
This would require moving and updating issue at the same time since "required field" can be anything. The JIra API supports this operation but its a bit of a mess to make it work correctly.

You can do something like below for now. It's not ideal but should work.
```sh
jira issue edit ISS-5516 --custom=comment="My comment" --no-input && jira issue move ISS-5516 "Done"
```

**Reply by azamora-zsc** [reply]
Thank you very much for your answer. However, I already tried what you suggest and it doesn't work (same error). When we do that, we are adding a comment in one action, and then attempt to transition the issue in another call to the API. Apparently, this should be done in an atomic way...
I understand your point about the required field (it could be anything). Maybe passing the field name and its value can make the trick:
```
jira issue move ISS-3245 "Done" --required-field comment --required-field-value "My comment about the transition"
```

**Reply by MattPD** [reply]
I think I have the same problem: Closing issues with required custom fields (all issues on my Jira instance) does not appear to be possible with `jira issue move`.

Context: My Jira instance requires a custom field when closing an issue.
In the Web UI I have to select one of the resolution types, https://support.atlassian.com/jira-cloud-administration/docs/what-are-issue-statuses-priorities-and-resolutions/
If I don't select any Resolution I cannot proceed (the error message in the Web UI: "The selected resolution is invalid.") I can proceed when I select, e.g., "Fixed" (I can then also supply a comment).

Currently, when I attempt to close an issue using either `jira issue move ISSUE-123 Done` or `jira issue move ISSUE-123 CLOSED` I can see the following error message:
```
$ jira issue move ISSUE-123
? Desired state: CLOSED

Error:
  - customfield_18343: Please enter the Workaround Steps for this fix.

jira: Received unexpected response '400 '.
Please check the parameters you supplied and try again.
```

"Workaround Steps" is just one of the several choices available in the Web UI, so I expect it gives up at this point.
That being said, I don't think I can currently provide any input to the required custom fields when using `jira issue move` (I know other commands support custom fields, but `jira issue move --help` does not seem to list any way to do that for `jira issue move`).

**Comment by ankitpokhrel** [comment, marked as answer]
Jira API allows to pass `resolution`, `comment` and `assignee` field in the endpoint. So,  the new release [v1.2.0](https://github.com/ankitpokhrel/jira-cli/releases/tag/v1.2.0) includes a change to allow you to set these fields when moving the issue.

```sh
$ jira issue move ISS-1 Done --comment "Example comment"
$ jira issue move ISS-1 Done -R Fixed
```

Note that these fields needs to be available in the Workflow to work, else they will be ignored.

See: https://confluence.atlassian.com/jirakb/how-to-add-a-comment-during-a-transition-779160682.html
